src/Inkluzitron/Modules/ImagesModule.cs

```diff
@@ -24,15 +27,82 @@ namespace Inkluzitron.Modules
     public class ImagesModule : ModuleBase
     {
         private ProfilePictureService ProfilePictureService { get; }
+        private BotDatabaseContext DbContext { get; set; }
+        private DatabaseFactory DatabaseFactory { get; }
 
-        public ImagesModule(ProfilePictureService profilePictureService)
+
+        public ImagesModule(ProfilePictureService profilePictureService, DatabaseFactory databaseFactory)
         {
+            DatabaseFactory = databaseFactory;
             ProfilePictureService = profilePictureService;
 
             if (!Directory.Exists("ImageCache"))
                 Directory.CreateDirectory("ImageCache");
         }
 
+        protected override void BeforeExecute(CommandInfo command)
+        {
+            DbContext = DatabaseFactory.Create();
+            base.BeforeExecute(command);
+        }
+
+        protected override void AfterExecute(CommandInfo command)
+        {
+            DbContext?.Dispose();
+            base.AfterExecute(command);
+        }
+
+        private static readonly double WhipBdsmTreshold = 0.5; // TODO config maybe?
+
+        /// <summary>
+        /// Validation based on BDSM results.
+        /// This check prevents sub users from whipping dom users
+        /// </summary>
+        /// <param name="source">Initiator (whipping user)</param>
+        /// <param name="target">Target (user to be whipped)</param>
+        /// <returns>If source user can whip target user</returns>
+        private async Task<bool> CanWhipUser(IUser source, IUser target)
+        {
+            // Users can always whip themselves
+            if (source.Id == target.Id) return true;
+
+            var tests = DbContext.BdsmTestOrgQuizResults
+                .Include(r => r.Items)
+                .OrderByDescending(r => r.SubmittedAt);
+
+            // Both users must have at least one BDSM test record
+            // Most recent test is used
+            var sourceTest = await tests.FirstOrDefaultAsync(r => r.SubmittedById == source.Id);
+            if (sourceTest == null) return true;
+
+            var targetTest = await tests.FirstOrDefaultAsync(r => r.SubmittedById == target.Id);
+            if (targetTest == null) return true;
+
+            var sourceTraits = sourceTest.Items.OfType<QuizDoubleItem>()
+                .Where(i => i.Value >= WhipBdsmTreshold);
+
+            var targetTraits = targetTest.Items.OfType<QuizDoubleItem>()
+                .Where(i => i.Value >= WhipBdsmTreshold);
+
+            var sourceIsDom = sourceTraits.Any(i => i.Key == "Dominant" || i.Key == "Master/Mistress");
```

Comment:
Radši bych tohle měl načtené z konfigurace, třeba jako seznam těch číselných ID. Nerad vidím podobné hardcoded.

Reply:
Od 02d6f60 už ot není hardcoded string, ale je to kinda hardcoded enum. Nicméně si myslím, že už je to v aktuálním stavu daleko hezčí.